cmd/consensus/main.go

```diff
@@ -238,92 +239,68 @@ func main() {
 			return err
 		}).
 		Module("random beacon key", func(node *cmd.NodeConfig) error {
-			// If this node was a participant in a spork, their DKG key for the
+			// If this node was a participant in a spork, their beacon key for the
 			// first epoch was generated during the bootstrapping process and is
 			// specified in a private bootstrapping file. We load their key and
 			// store it in the db for the initial post-spork epoch for use going
 			// forward.
+			//
 			// If this node was not a participant in a spork, they joined at an
-			// epoch boundary, so they have no DKG file (they will generate
-			// their first DKG private key through the procedure run during the
-			// current epoch setup phase), and we do not need to insert a key at
-			// startup.
+			// epoch boundary, so they have no beacon key file (they will generate
+			// their first beacon private key through the DKG in the EpochSetup phase
+			// prior to their first epoch as network participant).
 
-			// if the node is not part of the current epoch identities, we do
-			// not need to load the key
-			rootEpoch := node.State.AtBlockID(node.RootBlock.ID()).Epochs().Current()
-			initialIdentities, err := rootEpoch.InitialIdentities()
+			rootSnapshot := node.State.AtBlockID(node.RootBlock.ID())
+			isSporkRoot, err := protocol.IsSporkRootSnapshot(rootSnapshot)
 			if err != nil {
-				return err
+				return fmt.Errorf("could not check whether root snapshot is spork root: %w", err)
 			}
-			if _, ok := initialIdentities.ByNodeID(node.NodeID); !ok {
-				node.Logger.Info().Msg("node joined at epoch boundary, not reading DKG file")
+			if !isSporkRoot {
+				node.Logger.Info().Msg("node starting from mid-spork snapshot, will not read spork random beacon key file")
```

Comment:
is it possible that node joining from mid-spork, but still have random beacon key, is it worth to check?

Reply:
> is it possible that node joining from mid-spork, but still have random beacon key

It's possible the node operator has the beacon key file, but it cannot be valid for the root epoch (the node is not a random beacon committee member in that epoch). 

We could check for the file and warn or crash if we see it, as a likely misconfiguration, but we can't make use of it.

Reply:
OK, BTW, do we log the block height of the snapshot, if it's bootstrapped from a segment? 

Maybe we can log here.

Reply:
I actually just added this in a separate PR: https://github.com/onflow/flow-go/pull/1893/files#diff-cae89c2f7daa439631988089cdd51c551e4c25db2452885527785d0774235041R690-R694. With that, we will log the root and latest finalized height on startup.